src/tests/src/solver/expressions/test_CompareVisitor.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/unit_test.hpp>
+
+#include <antares/solver/expressions/Registry.hxx>
+#include <antares/solver/expressions/nodes/ExpressionsNodes.h>
+#include <antares/solver/expressions/visitors/CloneVisitor.h>
+#include <antares/solver/expressions/visitors/CompareVisitor.h>
+#include <antares/solver/expressions/visitors/PrintVisitor.h>
+
+using namespace Antares::Solver;
+using namespace Antares::Solver::Nodes;
+using namespace Antares::Solver::Visitors;
+
+// Only necessary for BOOST_CHECK_EQUAL
+
+BOOST_AUTO_TEST_SUITE(_CompareVisitor_)
+
+static Node* createSimpleExpression(Registry<Node>& registry, double param)
+{
+    Node* var1 = registry.create<LiteralNode>(param);
+    Node* param1 = registry.create<ParameterNode>("param1");
+    Node* expr = registry.create<AddNode>(var1, param1);
+    return expr;
+}
+
+BOOST_FIXTURE_TEST_CASE(comparison_to_self_simple, Registry<Node>)
```

Comment:
Renaming : **comparison_to_self_simple** ==> **compare_a_simple_node_to_itself**
Easier to read when unit test title is a short sentence.